designs/PMUITabSwitchRefactoring.md

```diff
@@ -0,0 +1,118 @@
+
+## Solution
+I believe we unnecessarily refresh the package lists on every tab-switch. Users can refresh their package data by using the Refresh button or performing any package action (ie, Install, Update, change Sources). 
+
+Therefore, showing and hiding lists that are maintained in memory results in a much better experience.
+
+### My branch
+* My current branch (still actively working on it): https://github.com/NuGet/NuGet.Client/tree/dev-donnie-msft-prototypeTwoItemsLists
+* Current commit on branch as of this writing: https://github.com/NuGet/NuGet.Client/commit/420e1d12fa418089f1f299b981d1ed409ee59cc4
+* _(I've included links throughout this spec to specific lines)_
+
+#### Active work
+* [Loading Status Indicator refactoring for new tab lists paradigm #10150](https://github.com/NuGet/Home/issues/10150)
+* [Updating in PMUI causes Object reference not set to an instance of an object #9882](https://github.com/NuGet/Home/issues/9882)
+* [Refresh button should work in new tabbing paradigm #10147](https://github.com/NuGet/Home/issues/10147)
+* [Store the search terms on each tab independently & Installed data is searched in UI #10148](https://github.com/NuGet/Home/issues/10148)
+* [Details Pane needs to reselect appropriate package #10145](https://github.com/NuGet/Home/issues/10145)
+* [Details Pane Versions Dropdown default selection should be Installed/LatestStable on Installed/Updates tabs #9887](https://github.com/NuGet/Home/issues/9887)
+
+### Core Concepts
+* From a user's point-of-view, there has traditionally been only one list of packages. My solution adds a **second Listbox** that I show and hide depending on selected tab. 
+
+* Additionally, a WPF [CollectionViewSource.Filter](https://docs.microsoft.com/en-us/dotnet/api/system.windows.data.collectionviewsource?view=netcore-3.1#events) ("UI Filtering") is applied to only show packages with Updates Available on the Updates tab.
+
+### High-level Logic
+`_listBrowse` - Represents packages on the Browse tab only
+
+`_listInstalled` - Represents packages on the Installed, Updates, and Consolidate tabs
+
+**Browse tab**
+* `_listBrowse` is Visibile https://github.com/NuGet/NuGet.Client/blob/420e1d12fa418089f1f299b981d1ed409ee59cc4/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml#L379
+* `_listInstalled` is Collapsed https://github.com/NuGet/NuGet.Client/blob/420e1d12fa418089f1f299b981d1ed409ee59cc4/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml#L388
+
+**Installed tab**
+* `_listBrowse` is Collapsed
+* `_listInstalled` is Visibile
+* UI Filter set to null 
+
+**Updates tab**
+* `_listBrowse` is Collapsed
+* `_listInstalled` is Visibile
+* UI Filter set to only packages with an Update Available https://github.com/NuGet/NuGet.Client/blob/420e1d12fa418089f1f299b981d1ed409ee59cc4/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs#L390
```

Comment:
Only part I worry is correctness/stale data issue. What if new version just pushed and available, so when we're going to see new latest updates? Even though current implementation is keep re-loading with every tab switch, still data is fresh and current.

Reply:
Right, we aggressively refresh, today. However, the worst case I can think of is a new Version of a package already in memory is available on a source, and that's not reflected in the UI. The natural user response to that is to press the "refresh" button.
I'm not sure if we really need a timer to automatically refresh.